src/App.js

```diff
@@ -7,66 +7,85 @@ import {
   NavLink,
   Redirect,
 } from 'react-router-dom';
+import { db } from './lib/firebase';
+import { useCollection } from 'react-firebase-hooks/firestore';
+import { SnackbarProvider } from 'notistack';
 
 import AddItems from './AddItems';
 import ItemList from './ItemList';
 import Welcome from './Welcome';
 
-
 function App() {
-  const hasToken = localStorage.getItem('userToken');
+  const [hasToken, setToken] = useState(null);
+
+  // This is listening to see if there there is `hasToken`
+  // Once we have a value, then we call `setToken` and are redirected to the list
+  useEffect(() => {
+    const user = localStorage.getItem('userToken');
+    user && setToken(user);
+  }, [hasToken]);
+
+  const [list, loading, error] = useCollection(
+    db.collection(hasToken || 'defaultValue'), // default value to prevent failure
+    {
+      snapshotListenOptions: { includeMetadataChanges: true },
+    },
+  );
+
   return (
```

Comment:
This  does not look easy! Good job on this. I'm excited for you to explain and talk about your process working on this.